src/hooks/useAnalytics/usePushAnalyticEvent.ts

```diff
@@ -11,21 +12,29 @@ export const usePushAnalyticEvent = () => {
     });
 
     const track = useCallback(
-        (options: URLSearchParams) =>
-            sendAnalytics?.({
-                body: options.toString(),
-                contentType: 'application/x-www-form-urlencoded',
-            }),
+        (options: URLSearchParams, componentName: ExternalComponentType) =>
+            sendAnalytics?.(
+                {
+                    body: options.toString(),
+                    contentType: 'application/x-www-form-urlencoded',
+                },
+                {
+                    query: {
+                        component: componentName,
+                    },
+                }
```

Comment:
Considering this previous [comment](https://github.com/Adyen/adyen-platform-experience-web/pull/788/files#r2439557434), I'd say to only add query params when `componentName` is provided.

```suggestion
                {...(componentName ? {
                    query: {
                        component: componentName,
                    },
                } : {})}
```